src/lib/gundb/UserStorageClass.js

```diff
@@ -1092,6 +1101,37 @@ export class UserStorage {
     return profile !== undefined
   }
 
+  /**
+   * Save survey
+   * @param {string} hash
+   * @param {object} details
+   * @returns {Promise<void>}
+   */
+  async saveSurveyDetails(hash: string, details: SurveyDetails) {
+    try {
+      await this.gun
+        .get('survey')
+        .get(hash)
+        .set(details)
```

Comment:
if you supply the hash there's no need for set. use .putAck if you use await.
you can do .get('survey').set(details) gundb would automatically generate some id ie hash.

lets also save it by date. DDMMYY/ get('survey').get(day)...